src/AssemblerReads.cpp

```diff
@@ -183,14 +184,90 @@ void Assembler::writeReadsSummary()
         csv << double(vertexCount) / double(markerCount) << ",";
 
         // Oxford Nanopore standard metadata.
-        csv << reads.getMetaData(readId, "runid") << ",";
-        csv << reads.getMetaData(readId, "sampleid") << ",";
-        csv << reads.getMetaData(readId, "read") << ",";
-        csv << reads.getMetaData(readId, "ch") << ",";
-        csv << reads.getMetaData(readId, "start_time") << ",";
+        csv << reads->getMetaData(readId, "runid") << ",";
+        csv << reads->getMetaData(readId, "sampleid") << ",";
+        csv << reads->getMetaData(readId, "read") << ",";
+        csv << reads->getMetaData(readId, "ch") << ",";
+        csv << reads->getMetaData(readId, "start_time") << ",";
 
         // End of the line for this read.
         csv << "\n";
-     }
+    }
+}
+
+
+void Assembler::adjustCoverage(uint64_t minReadLength, uint64_t desiredCoverage) {
+    if (desiredCoverage == 0) {
+        // Not specified. Skip.
+        return;
+    }
+
+    // Need to compute histograms to adjust coverage.
+    reads->computeReadLengthHistogram();
```

Comment:
With the code as written, the read length histogram `csv` files refer to the reads actually used by the assembler. This is consistent with current behavior. Perhaps it makes sense to also write separate `csv` files including all reads longer than `minReadLength`? To do this, we could call `Assembler::histogramReadLength` with a different `fileName` (e. g. `ExtendedReadLengthHistogram`) before calling `Assembler::adjustCoverage`. In that case, this call to `Reads::computeReadLengthHistogram` could be removed. 

Both kinds of histogram files would be useful. The ones including all reads longer than `minReadLength` could be used to study possible effects of different values of `minReadLength`.

But if `desiredCoverage` is 0, only one set of csv files should be written, like we do now.